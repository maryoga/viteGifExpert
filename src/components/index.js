// archivo de barril - referencia en GifExpertApp
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';