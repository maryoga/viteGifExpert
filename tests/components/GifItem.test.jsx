import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Test in <GifItem />', () => {

    const title = 'Saddie';
    const url = 'https://assetsnffrgf-a.akamaihd.net/assets/m/2015730/univ/art/2015730_univ_lsr_xl.jpg';

    test('Debe hacer match con el snapshot ', () => {
        const { container } = render(<GifItem title= { title } url= { url } />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title= { title } url= { url } />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url ); // lo evaluo por el 'URL'
        // Código más limpio, hago desestructuración de la imagen, usando el source 'src', y el texto alternativo 'alt'
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render( <GifItem title= { title } url= { url } />);
        expect( screen.getByText( title )).toBeTruthy(); // validar que el titulo exista
    });
});