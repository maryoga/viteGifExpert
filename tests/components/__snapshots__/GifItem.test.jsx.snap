// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem /> Debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saddie"
      src="https://assetsnffrgf-a.akamaihd.net/assets/m/2015730/univ/art/2015730_univ_lsr_xl.jpg"
    />
    <p>
      Saddie
    </p>
  </div>
</div>
`;
